agcaballero/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -35,144 +46,130 @@
-        case 0:
-          if(! availableWithOptional && optional) {
-            startWithOptional = i;
-            availableWithOptional = true;
-          }
-
-          if(! available) {
-            start = i;
-            available = true;
-          }
-
-          break;
-
-        case 1:
-          if(availableWithOptional && optional) {
-            addMeeting(request, startWithOptional, i, false, timesWithOptional);
-            availableWithOptional = false;
-          }
-          
-          if(! available) {
-            start = i;
-            available = true;
-          }
 
-          break;
+    for(int status = 0; status >= minStatus; status--) {
+      // an array list of available times 
+      ArrayList<TimeRange> availableTimes = (ArrayList<TimeRange>) findTimesForStatus(request, minuteStatus, status);
 
-        case 2: 
-          if(availableWithOptional && optional) {
-            addMeeting(request, startWithOptional, i, false, timesWithOptional);
-            availableWithOptional = false;
-          }
-          
-          if(available) {
-            addMeeting(request, start, i, false, times);
-            available = false;
-          }
-
-          break;
-      }
+      // as soon as the times are found return it, bc status only declines after this point
+      if(availableTimes.size() > 0) return availableTimes;
     }
 
-    // add meetings for end of day
-    if(availableWithOptional && optional) {
-      addMeeting(request, startWithOptional, TimeRange.END_OF_DAY, true, timesWithOptional);
-    }
-          
-    if(available) {
-      addMeeting(request, start, TimeRange.END_OF_DAY, true, times);
-    }
-    
-    // if there are any times where everyone can attend, return those
-    // else just return times where all mandatory attendees can come
-    if(timesWithOptional.size() > 0 || (! mandatory && optional)) {
-      return timesWithOptional;
-    }
-    else  {
-      return times;
-    }
+    // if there are no available times, return an empty array list
+    return new ArrayList<TimeRange>();
   }
   
   /**
-   * A private helper method to determine if there is overlap between two groups
-   * of attendees, represented as String collections
+   * A private helper method to determine the amount of overlap between
+   * two String Collections
    *
-   * @return true if overlap between attendees of two events, false otherwise
+   * @return the number of elements the two Collections have in common
    */
-  private boolean attendeeOverlap(Collection<String> groupA, Collection<String> groupB) {
-    for(String attendeeA: groupA)
-      for(String attendeeB: groupB)
-        if(attendeeA.equals(attendeeB)) return true;
+  private int amountOfOverlap(Collection<String> left, Collection<String> right) {
+    int overlap = 0;
 
-    return false;
+    for(String leftElement: left)
+      for(String rightElement: right)
+        if(leftElement.equals(rightElement))
+          overlap++;
```

Comment:
No, it isn't expected. I changed the strategy I used, so now this method should be accurate. The new strategy is just to preprocess left and right so that you have maps with the number of times an element occurs. Then, if an element is a key for both maps, you add the minimum value to overlap.

Reply:
Done